lib/filemaker/model/builder.rb

```diff
@@ -42,10 +30,15 @@ def single(resultset, klass)
 
         record.keys.each do |fm_field_name|
           # record.keys are all lowercase
-          field = klass.find_field_by_name(fm_field_name)
+          field = object.class.find_field_by_name(fm_field_name)
           next unless field
 
-          object.public_send("#{field.name}=", record[fm_field_name])
+          if hydrate_directly
+            # Because we are using ActiveModel::Dirty, so we hydrate directly.
+            object.attributes[field.name] = field.coerce(record[fm_field_name])
```

Comment:
is possible to get rid of this branching logic and just use 
`object.attributes[field.name] = field.coerce(record[fm_field_name])` instead of
`object.public_send("#{field.name}=", record[fm_field_name])`